pygmt/src/grdhisteq.py

```diff
@@ -0,0 +1,267 @@
+"""
+grdhisteq - Perform histogram equalization for a grid.
+"""
+
+import pandas as pd
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    args_in_kwargs,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+class grdhisteq:
+    r"""
+    Perform histogram equalization for a grid.
+
+    Two common use cases of :meth:`pygmt.grdhisteq` are to find data values
+    that divide a grid into patches of equal area using
+    :meth:`pygmt.grdhisteq.equalize_grid` or to write a grid with
+    statistics based on some kind of cumulative distribution function using
+    :meth:`pygmt.grdhisteq.compute_bins`.
+
+    Histogram equalization provides a way to highlight data that has most
+    values clustered in a small portion of the dynamic range, such as a
+    grid of flat topography with a mountain in the middle. Ordinary gray
+    shading of this grid (using :meth:`pygmt.Figure.grdimage` or
+    :meth:`pygmt.Figure.grdview`) with a linear mapping from topography to
+    graytone will result in most of the image being very dark gray, with the
+    mountain being almost white. :meth:`pygmt.grdhisteq.compute_bins` can
+    provide a list of data values that divide the data range into divisions
+    which have an equal area in the image [Default is 16 if ``divisions`` is
+    not set]. The :class:`pandas.DataFrame` or ASCII file output can be used to
+    make a colormap with :meth:`pygmt.makecpt` and an image with
+    :meth:`pygmt.Figure.grdimage` that has all levels of gray occuring
+    equally.
+
+    :meth:`pygmt.grdhisteq.equalize_grid` provides a way to write a grid with
+    statistics based on a cumulative distribution function. In this
+    application, the ``outgrid`` has relative highs and lows in the same
+    (x,y) locations as the ``grid``, but the values are changed to reflect
+    their place in the cumulative distribution.
+    """
+
+    @staticmethod
+    def _grdhisteq(grid, **kwargs):
```

Comment:
Should we put the `@use_alias` decorator here instead? This would 1) prevent duplication, and 2) allow us to set default values on the methods (e.g. `equalize_histogram(grid, outgrid=True, ...)` and `compute_bins(grid, outfile=True, ...)`.

Downside is that users won't have tab-completion to see what parameter options are available :slightly_smiling_face:

Reply:
I moved the `@use_alias` decorator to _grdhisteq. Rather than using `**kwargs`, I added positional arguments to the public methods so that tab completion still works. The downside is that single-character arguments do not work for these methods. I think this is fine since the only option that isn't included is **h**, which I would say is a bit unstable in PyGMT anyways. I would prefer to require long-form arguments where possible from the start rather than having more deprecations later on (xref https://github.com/GenericMappingTools/pygmt/issues/262).